text_to_sql.py

```
@@ -0,0 +1,172 @@
+            logger.error(f"No JSON found in response: {raw}")
+            raise ValueError("No valid JSON found in LLM response")
+
+        json_str = match.group()
+        logger.debug(f"Extracted JSON: {json_str}")
+
+        # Parse JSON
+        try:
+            data = json.loads(json_str)
+        except json.JSONDecodeError as e:
+            logger.error(f"JSON decode error: {e}, JSON: {json_str}")
+            raise ValueError(f"Invalid JSON in LLM response: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)